sdcm/scan_operation_thread.py

```diff
@@ -39,14 +40,14 @@ def __init__(self, db_cluster: [BaseScyllaCluster, BaseCluster], duration: int,
         self.log = logging.getLogger(self.__class__.__name__)
         self._thread = threading.Thread(daemon=True, name=self.__class__.__name__, target=self.run)
 
-    def get_ks_cs(self, db_node: BaseNode):
-        ks_cf_list = self.db_cluster.get_non_system_ks_cf_list(db_node)
-        if self.ks_cf not in ks_cf_list:
-            self.ks_cf = 'random'
-
-        if 'random' in self.ks_cf.lower():
-            return random.choice(ks_cf_list)
-        return self.ks_cf
+    def wait_until_user_table_exists(self, db_node, table_name: str = 'random', timeout_min: int = 20):
+        text = f'Waiting until {table_name} user table exists'
+        if table_name.lower() == 'random':
+            wait.wait_for(func=lambda: len(self.db_cluster.get_non_system_ks_cf_list(db_node)) > 0, step=60,
+                          text=text, timeout=60 * timeout_min, throw_exc=True)
+        else:
+            wait.wait_for(func=lambda: table_name in (self.db_cluster.get_non_system_ks_cf_list(db_node)), step=60,
+                          text=text, timeout=60 * timeout_min, throw_exc=True)
```

Comment:
may be instead of failing test with throw_exc if for some reason table was not created again after 60 * timeout_min, report error, and let test continue to run?

Reply:
I agree.